resolvers/event.js

```diff
@@ -304,8 +336,9 @@ const eventResolvers = {
       });
       return db.event.findByPk(args.id);
     },
-    deleteEvent: async (parent, args) =>
-      Promise.all([
+    deleteEvent: async (parent, args) => {
+      validators.validateRole([Roles.COMMANDER], validators.demoRole);
```

Comment:
I noticed that you've copy pasted this validator to a lot of these resolvers

Do you think there's a way of adding the validator to all these resolvers without copy-pasting? I just feel like this might not be ideal in terms of maintainability. In ordinary REST we might have middleware we attach to each endpoint; is there an alternative here?

Reply:
middleware is an interesting option, but since roles can be different for particular endpoints, wouldn't we still have to handle those specific cases? i agree it would be cleaner not to copy paste this so many times, but i dunno if the benefits of introducing middleware at this stage would outweigh the complexity

one could argue that this is more maintainable in one sense, but yeah in general it ain't great lol

Reply:
@JasonYG

Reply:
offline discussion - not worth investing in middleware at this point